source/s3_client.c

```diff
@@ -684,6 +683,7 @@ struct aws_s3_meta_request *aws_s3_client_make_meta_request(
                 .user_data = client,
                 .max_connections = aws_s3_client_get_max_active_connections(client, NULL),
                 .port = port,
+                .proxy_options = options->proxy_options,
```

Comment:
We don't need to make a deep copy of options as when the endpoint creates a connection manager, it will deep copy proxy_options to proxy_config. So we can simply pass it down to the connection manager.

Reply:
this sounds like potential for race condition bugs. 
creation is async, so if it returns before endpoints are created and user deletes options, well run into random bugs

Reply:
Fixed by copying proxy options to proxy config.